controller_manager:
  ros__parameters:
    update_rate: 100

    # publishes joint_states
    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster
      publish_rate: 50

    # accepts FollowJointTrajectory on /scaled_joint_trajectory_controller/follow_joint_trajectory
    scaled_joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController
      joints:
        - shoulder_pan_joint
        - shoulder_lift_joint
        - elbow_joint
        - wrist_1_joint
        - wrist_2_joint
        - wrist_3_joint
      state_publish_rate: 25
      action_monitor_rate: 10
      allow_partial_joints_goal: false
      constraints:
        goal_time: 0.6
        stopped_velocity_tolerance: 0.02
      gains:
        shoulder_pan_joint:  {p: 100.0, i: 0.0, d: 1.0}
        shoulder_lift_joint: {p: 100.0, i: 0.0, d: 1.0}
        elbow_joint:         {p: 100.0, i: 0.0, d: 1.0}
        wrist_1_joint:       {p: 50.0,  i: 0.0, d: 0.5}
        wrist_2_joint:       {p: 50.0,  i: 0.0, d: 0.5}
        wrist_3_joint:       {p: 20.0,  i: 0.0, d: 0.2}

